controllers/argocdagent/create-agent-config.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+# Copyright 2025 ArgoCD Operator Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+RECREATE="$1"
+
+export ARGOCD_AGENT_PRINCIPAL_NAMESPACE=argocd
+
+IPADDR=$(ip r show default | sed -e 's,.*\ src\ ,,' | sed -e 's,\ metric.*$,,')
+
+if ! test -x argocd-agentctl; then
+	echo "Please build argocd-agentctl first by running 'make cli'" >&2
```

Comment:
This message should probably be something more like 'Please ensure argocd-agentctl is on your path', since this repo (argocd-operator) won't be able to build `argocd-agentctl` via `make cli`